docs/custom-flows/saml-connections.mdx

```diff
@@ -1,163 +1,186 @@
-      </button>
-    </div>
-  );
-}
-
-export default App;
-```
-</CodeBlockTabs>
-
-To initiate a SAMLflow for a user that is already signed in, you can use the [`user.createExternalAccount(params)`](/docs/references/javascript/user/create-metadata#create-external-account) method, where `user` is a reference to the currently signed in user.
-
-```jsx
-user
-  .createExternalAccount({ strategy: strategy, redirectUrl: 'your-redirect-url' })
-  .then(externalAccount => {
-    // navigate to
-    // externalAccount.verification!.externalVerificationRedirectURL
-  })
-  .catch(err => {
-    // handle error
-  });
-```
-
-## SAML account transfer flow
-
-When a user initiates a SAML SSO verification during sign-in, or sign-up, it may sometimes be necessary to move the verification between the two flows.
-
-For example, if someone already has an account, and tries to go through the sign up flow with the same SAML account, they can’t perform a successful sign up again. Or, if someone attempts to sign in with their SAML credentials but does not yet have an account, they won’t be signed in to the account. For these scenarios, we have “account transfers.”
-
-<CodeBlockTabs type="framework" options={["JavaScript"]}>
-```js
-// Get sign-in and sign-up objects in the SAML callback page
-const { signIn, signUp } = window.Clerk.client;
-
-// If the user has an account in your application, but does not yet 
-// have a saml account connected, you can use the transfer method to 
-// forward that information.
-
-const userExistsButNeedsToSignIn =
-  signUp.verifications.externalAccount.status === "transferable" &&
-  signUp.verifications.externalAccount.error?.code ===
-    "external_account_exists";
-
-if (userExistsButNeedsToSignIn) {
-  const res = await signIn.create({ transfer: true });
-  if (res.status === "complete") {
-    window.Clerk.setActive({
-      session: res.createdSessionId,
-    });
-  }
-}
-
-// If the user has an existing saml account but does not yet exist as 
-// a user in your app, you can use the transfer method to forward that 
-// information.
-
-const userNeedsToBeCreated =
-  signIn.firstFactorVerification.status === "transferable";
-
-if (userNeedsToBeCreated) {
-  const res = await signUp.create({
-    transfer: true,
-  });
-  if (res.status === "complete") {
-    window.Clerk.setActive({
-      session: res.createdSessionId,
-    });
+2. Create a route at the URL that the `redirectUrl` param points to. It's recommended to name this route `/sso-callback`. This route should call the [`Clerk.handleRedirectCallback()`](/docs/references/javascript/clerk/handle-navigation#handle-redirect-callback) method or simply render the prebuilt [`<AuthenticateWithRedirectCallback/>`](/docs/components/control/authenticate-with-callback) component.
+
+The following example shows two files:
+
+1. The sign-in page where the user can start the SAML flow.
+2. The SSO callback page where the SAML flow is completed.
+
+<Tabs items={["Next.js"]}>
```

Comment:
Why wrap it in `<Tabs>` if there's only one item?

Reply:
so that users can clearly see it's a next.js example, and if we ever choose to add more frameworks, the structure is there. I like this practice.